homeassistant/components/homewizard_climate/climate.py

```diff
@@ -0,0 +1,226 @@
+"""Climate platform for homewizard_climate."""
+import logging
+import time
+
+from homewizard_climate_websocket.model.climate_device_state import (
+    HomeWizardClimateDeviceState,
+)
+from homewizard_climate_websocket.ws.hw_websocket import HomeWizardClimateWebSocket
+
+from homeassistant.components.climate import (
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    FAN_ON,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Todo."""
+    websockets = hass.data[DOMAIN][entry.entry_id]["websockets"]
+    entities = [HomeWizardClimateEntity(ws, hass) for ws in websockets]
+    async_add_entities(entities)
+
+
+class HomeWizardClimateEntity(ClimateEntity):
+    """Todo."""
+
+    def __init__(
+        self, device_web_socket: HomeWizardClimateWebSocket, hass: HomeAssistant
+    ) -> None:
+        """Todo."""
+        self._device_web_socket = device_web_socket
+        self._device_web_socket.set_on_state_change(self.on_device_state_change)
+        self._hass = hass
+        self._logger = logging.getLogger(
+            f"{__name__}.{self._device_web_socket.device.identifier}"
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID for this device."""
+        return f"{self._device_web_socket.device.type}_{self._device_web_socket.device.identifier}"
```

Comment:
This seems static, it can be added in `__init__` with `self._attr_unique_id`.

Reply:
Maybe this is also true for `name(self)`

Reply:
It is true for both cases and some other properties such as `fan_mode`, `fan_modes`, `swing_modes`, `hvac_modes`, etc.
I chose overriding the properties mostly for the sake of readability and cleanness, do you see an inherent advantage in defining them in `__init__` as `self._property`?

Reply:
If there's no clear benefit of having the `self._attr_` (I assume there isn't since any caller outside of the `ClimateEntity` and `Entity`, would hit the property method, and there shouldn't be a difference hitting the super class's property, which simply returns `self._attr` or the overriden property), I'd rather keep this as is for readability.

Otherwise, I'd have to move ~15 other static properties into the constructor as `self._attr`, which makes it more difficult to read.